### ISSUE TITLE
Smart Quotes need escaping

### ISSUE THREAD

**Issue by jborean93** [FriedrichWeinmann/ConvertToPsd1#2]
When emitting a quoted value the code currently escapes the ASCII single quotes correctly but it does not do it for the Unicode alternatives that are used as [a single quote equivalent](https://github.com/PowerShell/PowerShell/blob/2f4f585e7fe075f5c1669397ae738c554fa18391/src/System.Management.Automation/engine/parser/CharTraits.cs#L265-L272):

+ [U+2018](https://www.compart.com/en/unicode/U+2018)
+ [U+2019](https://www.compart.com/en/unicode/U+2018)
+ [U+201A](https://www.compart.com/en/unicode/U+201A)
+ [U+201B](https://www.compart.com/en/unicode/U+201B)

```powershell
$psd1 = @{foo = "Hello $([char]0x2018)World$([char]0x2018)" } | ConvertTo-Psd1

$psd1
# @{
#     foo = 'Hello ‘World‘'
# }


$psd1 | Invoke-Expression
# Invoke-Expression: Unexpected token 'World‘'' in expression or statement.
```

[CodeGeneration.EscapeSingleQuotedStringContent](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.language.codegeneration.escapesinglequotedstringcontent?view=powershellsdk-7.4.0) can be used to correctly escape all the quotes needed to place a value inside a single quoted string

```powershell
$value = "Hello $([char]0x2018)World$([char]0x2018)"
@"
@{
    foo = '$([System.Management.Automation.Language.CodeGeneration]::EscapeSingleQuotedStringContent($value))'
}
"@

# @{
#     foo = 'Hello ‘‘World‘‘'
# }
```

**Comment by FriedrichWeinmann** [comment]
Thanks for the catch and proposing a solution.
Fix implemented and released in v1.0.1

**Comment by jborean93** [comment]
Thanks for the speedy fix.
